src/axom/slam/policies/BivariateSetInterfacePolicies.hpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef SLAM_BivarSetIfacePolicies_HPP
+#define SLAM_BivarSetIfacePolicies_HPP
+
+#include "axom/slam/BivariateSet.hpp"
+#include "axom/slam/policies/InterfacePolicies.hpp"
+
+namespace axom
+{
+namespace slam
+{
+namespace policies
+{
+namespace detail
+{
+template <typename Set1 = slam::Set<>, typename Set2 = slam::Set<>>
+struct ConcreteBivariateSet
+{
+  using FirstSetType = Set1;
+  using SecondSetType = Set2;
+
+  using PositionType = typename FirstSetType::PositionType;
+  using ElementType = typename FirstSetType::ElementType;
+
+  using SubsetType = void;
+  using RangeSetType = typename RangeSet<PositionType, ElementType>::ConcreteSet;
+
+  static constexpr PositionType INVALID_POS = PositionType(-1);
+
+  /**
+   * \brief Constructor taking pointers to the two sets that defines the range
+   *        of the indices of the BivariateSet.
+   *
+   * \param set1  Pointer to the first Set.
+   * \param set2  Pointer to the second Set.
+   */
+  ConcreteBivariateSet(
+    const Set1* set1 = policies::EmptySetTraits<Set1>::emptySet(),
+    const Set2* set2 = policies::EmptySetTraits<Set2>::emptySet())
+    : m_set1(set1)
+    , m_set2(set2)
+  { }
+
+  /** \brief Size of the first set.   */
+  inline PositionType firstSetSize() const
+  {
+    return getSize<FirstSetType>(m_set1);
+  }
+  /** \brief Size of the second set.   */
+  inline PositionType secondSetSize() const
+  {
+    return getSize<SecondSetType>(m_set2);
+  }
+
+  /** \brief Returns pointer to the first set.   */
+  const FirstSetType* getFirstSet() const { return m_set1; }
```

Comment:
Does `getFirstSet()` follow the pattern in SLAM or should it be `firstSet()`?

Reply:
I'd be ok with either way (perhaps happier with the latter).
If we make a change, please be careful about consistency throughout slam.

Reply:
These match the names in the virtual `BivariateSet` class -- I'll change them to `firstSet()`/`secondSet()` in a future PR